--- jslint.vim.orig	2009-08-20 11:46:04.202893849 -0600
+++ jslint.vim	2009-08-20 11:45:33.096874637 -0600
@@ -1,3 +1,24 @@
+
+" Global Options
+"
+" Enable/Disable highlighting of errors in source.
+" Default is Enable
+" To disable the highlighting put the line
+" let g:JSLintHighlightErrorLine = 0 
+" in your .vimrc
+if !exists("g:JSLintHighlightErrorLine")
+	let g:JSLintHighlightErrorLine = 1 
+endif
+
+function! s:JSLintClear()
+  " Delete previous matches
+  if exists('b:errors')
+    for error in b:errors
+      call matchdelete(error)
+    endfor
+  endif
+endfunction
+
 function! s:JSLint(args) range
   cclose " Close quickfix window
   cexpr [] " Create empty quickfix list
@@ -46,7 +67,9 @@
       let b:has_errors = 1
       let l:line = b:parts[1] + (b:firstline - 1) " Get line relative to selection
       " Add line to match list
-      call add(b:errors, matchadd('Error', '\%' . l:line . 'l'))
+	  if g:JSLintHighlightErrorLine == 1
+		  call add(b:errors, matchadd('Error', '\%' . l:line . 'l'))
+	  endif
 
       " Store the error for an error under the cursor
       if l:line == line('.')
@@ -74,4 +97,6 @@
 command! -range JSLintLight <line1>,<line2>call s:JSLint("echo")
 " Highlight errors and open quick fix window
 command! -range JSLint <line1>,<line2>call s:JSLint("qf")
+" Un Highlight errors
+command! JSLintClear call s:JSLintClear()
 
